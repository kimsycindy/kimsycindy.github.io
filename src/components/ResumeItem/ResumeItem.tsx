import React from 'react'

interface ResumeItemProps {
  company: string
  subtitle: string
  position?: string
}

const ResumeItem = ({ company, subtitle, position }: ResumeItemProps) => (
  <p>
    <strong>{company}</strong> | {subtitle}
    {position && (
      <>
        <br />
        {position}
      </>
    )}
  </p>
)

export default ResumeItem
