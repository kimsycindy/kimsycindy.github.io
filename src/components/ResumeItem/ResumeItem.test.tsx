import React from 'react'
import renderer from 'react-test-renderer'

import ResumeItem from './ResumeItem'

describe('ResumeItem', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ResumeItem
          company="Mock Company"
          subtitle="Mock Subtitle"
          position="Mock Position"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly without position prop', () => {
    const tree = renderer
      .create(<ResumeItem company="Mock Company" subtitle="Mock Subtitle" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
