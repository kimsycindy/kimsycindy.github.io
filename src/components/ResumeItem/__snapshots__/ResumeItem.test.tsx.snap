// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeItem renders correctly 1`] = `
<p>
  <strong>
    Mock Company
  </strong>
   | 
  Mock Subtitle
  <br />
  Mock Position
</p>
`;

exports[`ResumeItem renders correctly without position prop 1`] = `
<p>
  <strong>
    Mock Company
  </strong>
   | 
  Mock Subtitle
</p>
`;
